### PR TITLE
Multiply by scalar

### PR DESCRIPTION
**Pull request by Bihaqo**

### CODE REVIEW

File: ttax/ops.py [hunk 1]
```diff
@@ -291,3 +294,26 @@ def are_batches_broadcastable(tt_a, tt_b):
...
+  if is_tt_object(a):
+    return _mul_by_scalar(a, b)
+  else:
+    return _mul_by_scalar(b, a)
+
+
+@tt_vmap(1)
+def _mul_by_scalar(tt, c):
+  cores = list(tt.tt_cores)
+  cores[0] = jnp.multiply(cores[0], c)
```
Comment (Bihaqo) [inline ttax/ops.py]: Up to you, but why not `c * cores[0]`?

File: ttax/ops_test.py [hunk 2]
```diff
@@ -122,6 +122,32 @@ def testAddBroadcasting(self):
...
+  def testMultiplyByScalar(self):
+    # Multiply batch of TT-tensor by scalar.
+    c = 4.5
+    rng = jax.random.PRNGKey(0)
+    dtype = jnp.float32
+    tt = random_.tensor(rng, (2, 1, 3, 4), tt_rank=[1, 2, 4, 3, 1],
+                        dtype=dtype)
+    res_actual1 = ops.full(ops.multiply(tt, c))
+    res_actual2 = ops.full(tt * c)
+    res_desired = jnp.multiply(ops.full(tt), c)
```
Comment (Bihaqo) [inline ttax/ops_test.py]: Again, why not `c * ops.full(tt)`?

File: ttax/ops_test.py [hunk 3]
```diff
@@ -122,6 +122,32 @@ def testAddBroadcasting(self):
...
 
+  def testMultiplyByScalar(self):
+    # Multiply batch of TT-tensor by scalar.
+    c = 4.5
+    rng = jax.random.PRNGKey(0)
+    dtype = jnp.float32
+    tt = random_.tensor(rng, (2, 1, 3, 4), tt_rank=[1, 2, 4, 3, 1],
+                        dtype=dtype)
+    res_actual1 = ops.full(ops.multiply(tt, c))
+    res_actual2 = ops.full(tt * c)
```
Comment (Bihaqo) [inline ttax/ops_test.py:133]: Can you also add `c * tt`?

File: ttax/utils.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import jax
+import jax.numpy as jnp
+
+from ttax.base_class import TTBase
+from ttax.base_class import TT
+from ttax.base_class import TTMatrix
+from ttax.compile import WrappedTT
+
+def is_tt_tensor(arg) -> bool:
+  return isinstance(arg, TT) or isinstance(arg, WrappedTT)
```
Comment (Bihaqo) [inline ttax/utils.py]: What if wrapperTT is wrapped around a TT-matrix? I think it should be `(isinstance(arg, WrappedTT) and not arg.is_tt_matrix)`

Comment (fasghq) [reply ttax/utils.py]: Do you mean `arg.tt.is_tt_matrix`?

File: ttax/ops.py [hunk 5]
```diff
@@ -179,6 +183,9 @@ def add(tt_a, tt_b):
   Raises
     ValueError if the arguments shapes do not coincide.
   """
+  tt_a = tt1.tt if isinstance(tt1, WrappedTT) else tt1
```
Comment (Bihaqo) [inline ttax/ops.py]: Let's put this logic inside wrapped TT? So that all wrapped logic is concentrated there.

File: ttax/compile.py [hunk 6]
```diff
@@ -50,6 +50,12 @@ def __mul__(self, other):
   def __matmul__(self, other):
     return ops.matmul(self, other)
 
+  def __add__(self, other):
+    return ops.add(self, other)
```
Comment (Bihaqo) [inline ttax/compile.py:54]: `ops.add(self.tt, other.tt)` 

(see comment below)

File: ttax/ops.py [hunk 7]
```diff
@@ -291,3 +298,40 @@ def are_batches_broadcastable(tt_a, tt_b):
...
+  if is_tt_object(a):
+    return _mul_by_scalar(a, b)
+  else:
+    return _mul_by_scalar(b, a)
+
+
+@tt_vmap(1)
+def _mul_by_scalar(tt, c):
+  if isinstance(tt, WrappedTT):
+    cores = list(tt.tt.tt_cores)
```
Comment (Bihaqo) [inline ttax/ops.py]: Likewise

File: ttax/ops.py [hunk 8]
```diff
@@ -291,3 +298,40 @@ def are_batches_broadcastable(tt_a, tt_b):
     else:
       return False
   return True
+
+
+def multiply(a, b):
+  """Returns the result of multiplication of so called
```
Comment (Bihaqo) [inline ttax/ops.py]: Not sure if it's a general thing, but our internal python style guide says that the first sentence of docstring should fit in the first line and end with a dot. Plus I wouldn't talk about wrapped TT in a docstrings: users don't need to know about those. Just something like "Elementwise multiplication of two TT objects or number times TT multiplication."

**Pull request merged**